tests/cypress/tests-e2e/print.cy.js

```diff
@@ -565,4 +568,294 @@ describe('Testing print', () => {
+
+        // format is WMTS|GET_CAPABILITIES_URL|LAYER_ID
+        const fakeWmtsLayerUrlId1 = `WMTS|${fakeWmtsGetCapUrl1}|${fakeWmtsLayerId1}`
+        const fakeWmtsLayerUrlId2 = `WMTS|${fakeWmtsGetCapUrl1}|${encodeExternalLayerParam(fakeWmtsLayerId2)}`
+        const fakeWmtsLayerUrlId3 = `WMTS|${fakeWmtsGetCapUrl2}|${fakeWmtsLayerId3}`
+        const fakeWmtsLayerUrlId4 = `WMTS|${fakeWmtsGetCapUrl2}|${fakeWmtsLayerId4}`
+
+        beforeEach(() => {
+            // WMS intercept URL 1
+            cy.intercept(
+                {
+                    url: `${fakeWmsBaseUrl1}**`,
+                    query: { REQUEST: 'GetMap', LAYERS: fakeWmsLayerId1 },
+                },
+                {
+                    fixture: '256.png',
+                }
+            ).as('externalWMSGetMap-1-layer-1')
+            cy.intercept(
+                {
+                    url: `${fakeWmsBaseUrl1}**`,
+                    query: { REQUEST: 'GetMap', LAYERS: fakeWmsLayerId2 },
+                },
+                {
+                    fixture: '256.png',
+                }
+            ).as('externalWMSGetMap-1-layer-2')
+            cy.intercept(
+                { url: `${fakeWmsBaseUrl1}**`, query: { REQUEST: 'GetCapabilities' } },
+                { fixture: 'external-wms-getcap-1.fixture.xml' }
+            ).as('externalWMSGetCap-1')
+
+            // WMS intercept URL 2
+            cy.intercept(
+                { url: `${fakeWmsBaseUrl2}**`, query: { REQUEST: 'GetMap' } },
+                {
+                    fixture: '256.png',
+                }
+            ).as('externalWMSGetMap-2')
+            cy.intercept(
+                { url: `${fakeWmsBaseUrl2}**`, query: { REQUEST: 'GetCapabilities' } },
+                { fixture: 'external-wms-getcap-2.fixture.xml' }
+            ).as('externalWMSGetCap-2')
+
+            // WMTS intercept URL 1
+            cy.intercept(`${fakeWmtsGetCapUrl1}`, {
+                fixture: 'external-wmts-getcap-1.fixture.xml',
+            }).as('externalWMTSGetCapOl-1')
+            cy.intercept(`${fakeWmtsGetCapUrl1}?**`, {
+                fixture: 'external-wmts-getcap-1.fixture.xml',
+            }).as('externalWMTSGetCap-1')
+
+            // WMTS intercept URL 2
+            cy.intercept(`${fakeWmtsGetCapUrl2}`, {
+                fixture: 'external-wmts-getcap-2.fixture.xml',
+            }).as('externalWMTSGetCapOl-2')
+            cy.intercept(`${fakeWmtsGetCapUrl2}?**`, {
+                fixture: 'external-wmts-getcap-2.fixture.xml',
+            }).as('externalWMTSGetCap-2')
+
+            cy.intercept(
+                'http://test.wmts.png/wmts/1.0.0/TestExternalWMTS-*/default/ktzh/**/*/*.png',
+                {
+                    fixture: '256.png',
+                }
+            ).as('externalWMTS')
+        })
+
+        it('prints external WMS correctly', () => {
+            interceptPrintRequest()
+            interceptPrintStatus()
+            interceptDownloadReport()
```

Comment:
Could also be moved to the `beforeEach` hook, as there's no test not needing that in this test file